### ISSUE TITLE
Failure to construct `TouchEvent` for very first `touchstart` event results in `NaN` or `undefined` offset values in `react-dnd`

### ISSUE THREAD

**Issue by yongla** [LouisBrunner/dnd-multi-backend#100]
## Context

An interesting bug that I've run into using `dnd-multi-backend` and `react-dnd`, is that `react-dnd`'s monitors can sometimes return `XYCoord` objects like `{ x: undefined, y: undefined }` or `{ x: NaN, y: NaN }`, when calling `monitor.getInitialClientOffset()`, `monitor.getDifferenceFromInitialOffset()`, etc. From going through `react-dnd`'s source code and their typings, I don't think this should never happen in theory.

This bug would also only happen when working with touch events (on actual touch devices, or even through Chrome responsive mode which simulates touch events).

## The error in `dnd-multi-backend`

From going through various stack traces, my current theory is that this issue in `react-dnd` happens because of this error that is thrown by `dnd-multi-backend`, which it tries to silently and gracefully handle:

```
Failed to construct 'TouchEvent': Please use the 'new' operator, this DOM object constructor cannot be called as a function.
```

Code reference: https://github.com/LouisBrunner/dnd-multi-backend/blob/main/packages/dnd-multi-backend/src/MultiBackendImpl.ts#L199-L204

Because of this error, and how it affects logic in `react-dnd`, use cases of using `react-dnd` where one might need to rely on offsets returned by monitors (e.g. custom drag layer) can break due to this error as the fallback to return a generic `Event` filters out a lot of important event attributes that might be needed.

## To reproduce

The error is reproducible on the `dnd-multi-backend` example page here: https://louisbrunner.github.io/dnd-multi-backend/examples/dnd-multi-backend.html.

<img width="1920" alt="Screen Shot 2022-12-07 at 7 37 04 AM" src="https://user-images.githubusercontent.com/6947911/206223183-5e1ca1c4-9377-4359-b9dd-0e6d46b399a2.png">

It only happens on the **very first** `touchstart` event. For some reason, the code above I refer to never seems to be invoked afterwards.

Note: Although I'm using simulated touches in the screenshot above, I can reproduce this on an iPad as well in Chrome and Safari.

iPad Safari reproduction example:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/6947911/206255562-72f33301-9ccb-48a1-8761-f89dc0485cb0.png">

## Workaround

Currently, I'm using `yarn patch` to manually determine the event class to invoke `new` on.

So, instead of `newEvent = event.constructor(event.type, event);`, I have:
```js
const EventForType = event.type.startsWith('touch')
  ? TouchEvent
  : event.type.startsWith('mouse')
  ? MouseEvent
  : event.type.startsWith('drag')
  ? DragEvent
  : event.type.startsWith('pointer')
  ? PointerEvent
  : Event;
newEvent = new EventForType(event.type, event);
```

This might be a bit hacky and I'm not too familiar with re-constructing events, so not sure if this would be the elegant solution to propose for this package, but wanted to share something that seems to work as a workaround for now.

---

Also, thanks for all your work on this project. It's very useful and I appreciate all the documentation and examples you provide as well.

**Comment by LouisBrunner** [comment]
Fix has been released as `7.1.3`, thanks again for diagnosticating this old bug! 😄

**Comment by yongla** [comment]
Amazing! Thanks for the very quick responsiveness on this 🔥
